airbyte-commons-worker/src/main/java/io/airbyte/workers/internal/AirbyteMessageTracker.java

```diff
@@ -53,13 +55,20 @@ public class AirbyteMessageTracker implements MessageTracker {
   private final BiMap<AirbyteStreamNameNamespacePair, Short> nameNamespacePairToIndex;
   private final Map<Short, Long> streamToTotalBytesEmitted;
   private final Map<Short, Long> streamToTotalRecordsEmitted;
+  private final Map<Short, Long> streamToTotalBytesEstimated;
+  private final Map<Short, Long> streamToTotalRecordsEstimated;
   private final StateDeltaTracker stateDeltaTracker;
   private final StateMetricsTracker stateMetricsTracker;
   private final List<AirbyteTraceMessage> destinationErrorTraceMessages;
   private final List<AirbyteTraceMessage> sourceErrorTraceMessages;
   private final StateAggregator stateAggregator;
   private final boolean logConnectorMessages = new EnvVariableFeatureFlags().logConnectorMessages();
 
+  // These variables support SYNC level estimates and are meant for sources where stream level
+  // estimates are not possible e.g. CDC sources.
+  private Long totalRecordsEstimatedSync;
```

Comment:
Looks like we are adding a new concept of Sync vs Stream, instead, would it make sense to use STREAM vs GLOBAL?

Reply:
I'm following from the AirbyteTraceEstimateMessage type. Slight preference to keep the names consistent though happy to change the name within this file to Global.

With the AirbyteTraceEstiamteMessage, I do think STREAM vs SYNC are decent names for now.

Reply:
I am thinking about consistency for the near future, Sync feels very coupled to how we current run the replication. For STREAM, I think we want to modify our current replication model to be able to replicate data at stream level rather than connection. If we are running per stream replication in parallel, Sync mode for progress looks outdated.

It probably feels more like future-proofing, I don't have reasons to feel strongly either way at this moment.

Reply:
Agree with your thoughts. Honestly, not 100% how the stream-as-first-class citizen refactor will play with CDC w.r.t the stats here. 

In that case, I'm guessing we will be able to do away with the SYNC field. I'm inclined to keep this as is for now and revisit when we cross that. How does that sound?

Reply:
Sounds good to me, conceptually, I think we are tracking the right way, we can address the naming later on.